### PR TITLE
fix: #49 allow newer version of dependencies

### LINKED ISSUE #49: [BUG] requirements pinned to rigidly
**Describe the bug**
jake is a build tool, and it isn't the only one. jake must be installed in the same venv as the packages it is trying to check, same as other tools like pytest. (I verified jake has to be installed in the same venv as the code under test, if I install to pipx & run jake, it will tell me about jake's dependencies). If a tool that must share the same venv as other tools pins all of its dependencies to an exact version (e.g. ==2.1.1) then you get conflicts. Yes, you can ignore conflicts until something comes along, I don't know says a vulnerability a package & you have update that package to the secure one.

`pip check`
jake 0.2.65 has requirement idna==2.10, but you have idna 2.8.
jake 0.2.65 has requirement requests==2.25.0, but you have requests 2.22.0.
jake 0.2.65 has requirement urllib3==1.26.2, but you have urllib3 1.25.11.

I recommend you change your requirements to something more like  library>=1.5.0

This won't make the problem go away, but it makes dependency hell more manageable, especially over time as the other tools in the ecosystem move on to higher dependency version numbers, sometimes before jake does.

### PR DESCRIPTION
**Pull request by bhamail**
Allow newer version of dependencies.

The `click` library had failures when the very recently released v8.0.0 was used, so I added an upper bound to solve that issue for now.

Also fixed an error due to db.purge deprecation, replaced with db.truncate call.

* Fixes #49

cc @bhamail / @DarthHater

### CODE REVIEW

**Comment by bhamail** [comment]
@jwa5426 Please holler if this looks sane to you.

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-pylint==2.6.0
-responses==0.12.0
-unittest-xml-reporting==3.0.4
+pylint>=2.6.0
+responses>=0.12.0
+unittest-xml-reporting>=3.0.4
```
Comment (DarthHater) [inline requirements-dev.txt:3]: Oof, I hate seeing stuff like this. Why do we have to do it? We can't just pin to a known version?

**Comment by jwa5426** [comment]
@bhamail I agree with preventing the click 8.0.0 issue, I actually ran into that as well. For some of the other non-issue ones maybe it is a good idea to keep them pinned to known good versions or at least a tighter range.

File: requirements-dev.txt (hunk 1 above)
Comment (bhamail) [reply requirements-dev.txt:3]: Seems fine to pin to specific dev deps, as I'm guessing that will not interfere with Jake being used to scan projects. I will push some changes to lock these dev dep version down (as they used to be).

**Comment by bhamail** [comment]
@jwa5426 Thanks for looking this over! I was trying to fix the issues described in Issue #49, and "looser" version ranges seemed better for the reasons mentioned there. That said, I'm not against add upper bounds to version ranges when we slam into known issues (e.g. `click 8.0.0`).

**Comment by jwa5426** [comment]
@bhamail seems reasonable to me 👍

File: requirements-dev.txt (hunk 1 above)
Comment (bhamail) [reply requirements-dev.txt]: Dev deps locked: 5189624

**Pull request merged**